inspirehep/modules/refextract/tasks.py

```diff
@@ -73,19 +74,55 @@ def create_journal_kb_file():
             (r.json -> '_collections')::jsonb ? 'Journals'
     """)
 
-    with codecs.open(refextract_journal_kb_path, encoding='utf-8', mode='w') as fd:
-        for row in titles_query:
-            normalized_short_title = _normalize(row['short_title'])
-            if normalized_short_title:
-                fd.write(u'{}---{}\n'.format(normalized_short_title, row['short_title']))
-            normalized_journal_title = _normalize(row['journal_title'])
-            if normalized_journal_title:
-                fd.write(u'{}---{}\n'.format(normalized_journal_title, row['short_title']))
-
-        for row in title_variants_query:
-            normalized_title_variant = _normalize(row['title_variant'])
-            if normalized_title_variant:
-                fd.write(u'{}---{}\n'.format(normalized_title_variant, row['short_title']))
+    encode = codecs.getencoder(encoding='utf-8')
+
+    kb_data = []
+    batch_size = 1000
+    for row in titles_query:
+        normalized_short_title = _normalize(row['short_title'])
+        if normalized_short_title:
+            encoded_short_title, _ = encode(
+                u'{}---{}\n'.format(
+                    normalized_short_title,
+                    row['short_title'],
+                )
+            )
+            kb_data.append(encoded_short_title)
+
+        normalized_journal_title = _normalize(row['journal_title'])
+        if normalized_journal_title:
+            encoded_journal_title, _ = encode(
+                    u'{}---{}\n'.format(
+                        normalized_journal_title,
+                        row['short_title'],
+                    )
+            )
+            kb_data.append(encoded_journal_title)
+
+        if len(kb_data) == batch_size:
+            with fsopen(refextract_journal_kb_path, mode='wb') as fd:
+                fd.write(''.join(kb_data))
+            kb_data = []
```

Comment:
I would suggest putting this code in a function since the pattern used is the same in three points.
Something like
```
. . . 
    for row in titles_query:
        _add_normalized_title(kb_data, row['short_title'])
        _add_normalized_title(kb_data, row['journal_title'])
        _flush_kb_data(batch_size, kb_data, refextract_journal_kb_path)
. . .